hyperspy/_components/polynomial.py

```diff
@@ -119,8 +122,18 @@ def estimate_parameters(self, signal, x1, x2, only_current=False):
                 # Shape needed to fit coefficients.map:
                 cmap_shape = nav_shape + (self.get_polynomial_order() + 1, )
                 self.coefficients.map['values'][:] = cmaps.reshape(cmap_shape)
-                if binned is True:
+                if self.binned:
                     self.coefficients.map["values"] /= axis.scale
                 self.coefficients.map['is_set'][:] = True
             self.fetch_stored_values()
             return True
+
+    def function_nd(self, axis):
```

Comment:
Because the `coefficients` parameter is defined as an array of several parameters, I am not sure if rewriting the polynomial as an Expression will be a good way to improve it. There may be better approach.

Reply:
Yes, that's the other problematic component that I've forgotten. However, see #1989

Reply:
Happy to see polynomial expression being done in #1989!! ;)